test/basic.js

```diff
@@ -65,6 +65,24 @@ test('subscribe QoS 0', function (t) {
   })
 })
 
+test('test empty subscriptions', function (t) {
+  t.plan(0)
+
+  var broker = aedes()
+  var s = connect(setup(broker))
+
+  s.inStream.write({
+    cmd: 'subscribe',
+    messageId: 24,
+    subscriptions: []
+  })
+
+  s.outStream.once('data', function (packet) {
+    t.fail('message received')
+  })
+  broker.on('closed', t.end.bind(t))
+})
```

Comment:
Are you sure this test is actually running? Who will emit ‘closed’?

Reply:
The closed event is from https://github.com/mcollina/aedes/blob/master/test/helper.js#L22

Reply:
Tried the following, we will receive `broker closed` message.
```
test('test empty subscriptions', function (t) {
  t.plan(0)

  var broker = aedes()
  var s = connect(setup(broker))

  s.inStream.write({
    cmd: 'subscribe',
    messageId: 24,
    subscriptions: []
  })

  s.outStream.once('data', function (packet) {
    t.fail('message received')
  })
  broker.on('closed', function () {
    console.log('broker closed')
    t.end()
  })
})
```

Reply:
I would prefer to not rely on a `setTimeout`. Would you mind restructuring the test in that way? A good way would be to send a DISCONNECT packet or similar, and check if the number of packet do not inclue a SUBACK.

BTW, it this behavior spec-compliant?